### PR TITLE
More robust service scripts

### PR DESCRIPTION
**Pull request by g2flyer**
Address issue #182 and #164 :
- start detects already running service(s) and returns only
  either after service(s) are up and running or a failure
 - stop does work also with non-default ports and config-files

- fixes small bug in docker Makefile where wrong environment file was sourced
  (omission from commit 7e4f29fe6883bfb43b1d4cf339ff1c8c66c35103)

### CODE REVIEW

**Comment by cmickeyb** [comment]
found a problem... 
pgrep finds processes that are not services because the command name includes the string. for example, ss-status.sh says that the processes are running because it found the process /usr/lib/accountsservice/accounts-daemon.

You can fix this pretty easily by replacing 'grep "$1"' in common.sh with 'egrep "\b$1\b"'

**Comment by g2flyer** [comment]
> found a problem...
> pgrep finds processes that are not services because the command name includes the string. for example, ss-status.sh says that the processes are running because it found the process /usr/lib/accountsservice/accounts-daemon.
> 
> You can fix this pretty easily by replacing 'grep "$1"' in common.sh with 'egrep "\b$1\b"'

good catch (thanks to trusty docker which has only minimal stuff running i didn't run into this problem test :-).    Anyway, pushed now a change which has a slighly more elaborate pattern to also capture basename/identity in addition to [esp]service ..

File: pservice/bin/ps-status.sh [hunk 1]
```diff
@@ -14,7 +14,31 @@
...
+source ${F_SERVICEHOME}/bin/common.sh # from ../../eservice/bin/
+
+F_USAGE='-b|--base name'
+F_BASENAME='pservice'
+
+# -----------------------------------------------------------------
+# Process command line arguments
+# -----------------------------------------------------------------
+TEMP=`getopt -o b:h --long base:,help \
+     -n 'stop.sh' -- "$@"`
```
Comment (bvavala) [inline pservice/bin/ps-status.sh]: do you want to shut down the services if the options are wrong?

File: eservice/bin/ss-status.sh [hunk 2]
```diff
@@ -14,7 +14,31 @@
...
+source ${F_SERVICEHOME}/bin/common.sh
+
+F_USAGE='-b|--base name'
+F_BASENAME='sservice'
+
+# -----------------------------------------------------------------
+# Process command line arguments
+# -----------------------------------------------------------------
+TEMP=`getopt -o b:h --long base:,help \
+     -n 'stop.sh' -- "$@"`
```
Comment (bvavala) [inline eservice/bin/ss-status.sh]: 1) again, I don't think the idea here is to shut down the services if there is any error
2) stop.sh should be the name of the program called by getopt ... but where is it defined?

File: pservice/bin/ps-status.sh (hunk 1 above)
Comment (g2flyer) [reply pservice/bin/ps-status.sh]: nope, but that's also not what happens.  This is only the string which is printed out as program identifier when you provide wrong arguments. But you definite point a bug which i inherited from before but might as well also just fix ...

Comment (g2flyer) [reply pservice/bin/ps-status.sh]: just pushed fix now ...

File: eservice/bin/ss-status.sh (hunk 2 above)
Comment (g2flyer) [reply eservice/bin/ss-status.sh]: see above

File: eservice/bin/es-stop.sh [hunk 3]
```diff
@@ -14,29 +14,40 @@
...
-for port in `seq $F_BASE $(($F_BASE+$F_COUNT))` ; do
-    curl -s "http://localhost:$port/shutdown"
+for index in `seq 1 $F_COUNT` ; do
+    IDENTITY="${F_BASENAME}$index"
+    echo stopping enclave service $IDENTITY
+
+    url="$(get_url_base $IDENTITY)/shutdown" || { echo "no url found for enclave service"; exit 1; }
+    resp=$(curl -sL -w "%{http_code}\\n" ${url} -o /dev/null)
+    if [ $? != 0 ]; then # shutdown results in 500 error, so do not test also || [ $resp != "200" ]
+	echo "enclave service $IDENTITY not running or not properly shut down"
```
Comment (prakashngit) [inline eservice/bin/es-stop.sh]: if shutdown of one service fails, isnt hutdown attempted for the remaining ones?

Comment (g2flyer) [reply eservice/bin/es-stop.sh]: i guess you are right, will change to continue (but still make sure there is a non-zero exit code ...

> if shutdown of one service fails, isnt hutdown attempted for the remaining ones?
i guess you are right, will change to continue (but still make sure there is a non-zero exit code ...

Comment (g2flyer) [reply eservice/bin/es-stop.sh]: Corresponding change is pushed now to PR ...

**Pull request merged**